sdk/vcs/git/git_tag.go

```diff
@@ -23,38 +23,53 @@ type TagOpts struct {
 
 // Tag makes git tag action
 func Tag(repo string, auth *AuthOpts, opts *TagOpts, output *OutputOpts) error {
-	if strings.HasPrefix(repo, "http://") || strings.HasPrefix(repo, "ftp://") || strings.HasPrefix(repo, "ftps://") {
-		return fmt.Errorf("Git protocol not supported")
-	}
-	if strings.HasPrefix(repo, "https://") {
-		return gitTagOverHTTPS(repo, auth, opts, output)
+	var commands []cmd
+	repoURL, err := getRepoURL(repo, auth)
+	if err != nil {
+		return err
 	}
-	return gitTagOverSSH(auth, opts, output)
+	commands = gitTagCommand(repoURL, opts)
+	return gitTagOver(repo, commands, auth, output)
 }
 
-func gitTagOverHTTPS(repo string, auth *AuthOpts, opts *TagOpts, output *OutputOpts) error {
-	if auth == nil {
-		cmd := gitTagCommand(opts)
-		return runCommand(cmd, output)
-	}
-	u, err := url.Parse(repo)
+// TagList List tag from given git directory
+func TagList(repo, dir string, auth *AuthOpts, output *OutputOpts) error {
+	repoURL, err := getRepoURL(repo, auth)
 	if err != nil {
 		return err
 	}
+	commands := gitTagListCommand(repoURL, dir)
+	return gitTagOver(repo, commands, auth, output)
+}
 
-	u.User = url.UserPassword(auth.Username, auth.Password)
+func getRepoURL(repo string, auth *AuthOpts) (string, error) {
+	if strings.HasPrefix(repo, "http://") || strings.HasPrefix(repo, "ftp://") || strings.HasPrefix(repo, "ftps://") {
+		return "", fmt.Errorf("Git protocol not supported")
+	}
+	if strings.HasPrefix(repo, "https://") {
+		u, err := url.Parse(repo)
+		if err != nil {
+			return "", err
+		}
+		u.User = url.UserPassword(auth.Username, auth.Password)
+		return u.String(), nil
+	}
+	return repo, nil
+}
 
-	cmd := gitTagCommand(opts)
-	return runCommand(cmd, output)
+func gitTagOver(repo string, commands []cmd, auth *AuthOpts, output *OutputOpts) error {
+	if strings.HasPrefix(repo, "https://") {
+		return runCommand(commands, output)
+	}
+	return gitTagOverSSH(commands, auth, output)
 }
 
-func gitTagOverSSH(auth *AuthOpts, opts *TagOpts, output *OutputOpts) error {
+func gitTagOverSSH(commands []cmd, auth *AuthOpts, output *OutputOpts) error {
```

Comment:
gitTagOverSSH make a gittag over ssh... well, we doesn't have to pass commands in parameter